erica/elster_xml/grundsteuer/elster_data_representation.py

```diff
@@ -3,15 +3,146 @@
 
 from erica.elster_xml.common.basic_xml_data_representation import ENutzdaten, construct_basic_xml_data_representation
 from erica.elster_xml.common.elsterify_fields import elsterify_anrede, elsterify_date
-from erica.request_processing.erica_input.v2.grundsteuer_input import Person, GrundsteuerData, \
-    Eigentuemer as EigentuemerInput, Anteil, Vertreter, Empfangsbevollmaechtigter
+from erica.request_processing.erica_input.v2.grundsteuer_input import GrundsteuerData
+from erica.request_processing.erica_input.v2.grundsteuer_input_eigentuemer import Vertreter, Anteil, Person, \
+    Empfangsbevollmaechtigter, Eigentuemer as EigentuemerInput
+from erica.request_processing.erica_input.v2.grundsteuer_input_gebaeude import Gebaeude as GebaeudeInput, \
+    WeitereWohnraeumeDetails, GaragenAnzahl
 
 """
     The content of the Grundsteuer Nutzdaten XML as its data prepresentation.
     The classes are prefixed with "E" for "Elster".
 """
 
 
+@dataclass
+class EGaragen:
+    E7403171: int  # Anzahl Garagen
+
+    def __init__(self, garagen_anzahl: GaragenAnzahl):
+        self.E7403171 = garagen_anzahl.anzahl_garagen
+
+
+@dataclass
+class EWohnUnter60:
+    E7403131: int = 0  # Anzahl
+    E7403132: int = 0  # Flaeche
+
+    def __init__(self, flaechen: List[int]):
+        for flaeche in flaechen:
+            if flaeche < 60:
+                self.E7403131 += 1
+                self.E7403132 += flaeche
+
+
+@dataclass
+class EWohn60bis100:
+    E7403141: int = 0
+    E7403142: int = 0
+
+    def __init__(self, flaechen: List[int]):
+        for flaeche in flaechen:
+            if 60 <= flaeche < 100:
+                self.E7403141 += 1
+                self.E7403142 += flaeche
+
+
+@dataclass
+class EWohnAb100:
+    E7403151: int = 0
+    E7403152: int = 0
+
+    def __init__(self, flaechen: List[int]):
+        for flaeche in flaechen:
+            if flaeche >= 100:
+                self.E7403151 += 1
+                self.E7403152 += flaeche
+
+
+@dataclass
+class EWeitereWohn:
+    E7403121: int = 0
+    E7403122: int = 0
+
+    def __init__(self, weitere_wohnraeume_details: WeitereWohnraeumeDetails):
+        self.E7403121 = weitere_wohnraeume_details.anzahl
+        self.E7403122 = weitere_wohnraeume_details.flaeche
+
+
+@dataclass
+class EAngDurchschn:
+    Wohn_Unter60: Optional[EWohnUnter60] = None
+    Wohn_60bis100: Optional[EWohn60bis100] = None
+    Wohn_ab100: Optional[EWohnAb100] = None
+    Weitere_Wohn: Optional[EWeitereWohn] = None
+
+    def __init__(self, input_data: GebaeudeInput):
+        flaechen: List[int] = []
+        if input_data.wohnflaeche:
+            flaechen.append(input_data.wohnflaeche.wohnflaeche)
+        elif input_data.wohnflaechen:
+            flaechen.append(input_data.wohnflaechen.wohnflaeche1)
+            flaechen.append(input_data.wohnflaechen.wohnflaeche2)
```

Comment:
True, and this is not the only case where we can optimize erica input. However, in order for this to work, we would need to add yet another transformation stage in grundsteuer which would introduce additional complexity and become a potential source of bugs. Personally, I would leave that for now and maybe revisit it once we start to implement validation.

Reply:
I thought *you* were always the person to tell me that we can optimize the input and have the translation layer anyways. 😂 Plus, just basically pasting this code in the webapp can spare us quite some work in erica going forward (introducing more wohnungen for example). So this to me is a very needed translation. (Btw, we already ned a translation layer anyway, because we need to translate "true"/"false" to boolean.

Reply:
Talked about it, we'll revisit in case we have time.